parameter.go

```diff
@@ -0,0 +1,178 @@
+package taskflow
+
+import (
+	"errors"
+	"strconv"
+)
+
+type parameter struct {
```

Comment:
it would be good to add a comment here. especially that reading the code it is hard to distinguish the difference between `param` and `parameter` type

Reply:
I will skip it for now. I have an idea of how to redesign the parameter registration. I want to make a draft PR for it first.

Reply:
Refactored 😄